connector_magento/models/product/common.py

```diff
@@ -108,12 +108,10 @@ def recompute_magento_qty(self):
         informations for each product.
         """
         # group products by backend
-        backends = defaultdict(self.browse)
-        for product in self:
-            backends[product.backend_id] |= product
-
-        for backend, products in backends.iteritems():
-            self._recompute_magento_qty_backend(backend, products)
+        for backend in self.mapped('backend_id'):
+            self._recompute_magento_qty_backend(
+                backend, self.filtered(
+                    lambda x: x.backend_id.id == backend.id))
```

Comment:
Yes but itering on a recordset is quicker than creating a new big one.

I have more than 26.000 products binded. The original code took more than 15 minutes to complete the dictionnary.

This version takes only 6 secondes ...